### ISSUE TITLE
Investigate and resolve convertToType usage in QueryUnbufferedAsync

### ISSUE THREAD

**Issue by mgravell** [DapperLib/Dapper#1920]
see review comment and discussion [here](https://github.com/DapperLib/Dapper/pull/1912/files/7ea1ab3adebd3c18ae6dd2355a2d1533140f5e93#r1228366767)

**Comment by mellis-ao** [comment]
I was looking at the `QueryUnbuffered` method today and noticed the unused `convertToType` this which lead me to check here if it was being looked at. I also noticed it exists in `QueryAsync` also:

https://github.com/DapperLib/Dapper/blob/bd4f75b512de3e00f2c2631d5309961a1ecfea23/Dapper/SqlMapper.Async.cs#L451

Reading the original discussion it does look like it would be safe to delete but I can't help but feel it would be a performance improvement to use it and stop `GetValue` from running that code for every result row:

https://github.com/DapperLib/Dapper/blob/bd4f75b512de3e00f2c2631d5309961a1ecfea23/Dapper/SqlMapper.cs#L1394

Apologies if I am bumping an old issue.

**Comment by mgravell** [comment]
Yes, I should revisit this. Note that the Dapper AOT pieces have a new attribute that takes this much further: `[StrictTypes]`. I should also further document that side of things!
